// anujs/aaa/dist/pages/demo/syntax/test/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _ReactWX = require('../../../../ReactWX');

var _ReactWX2 = _interopRequireDefault(_ReactWX);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function P() {
    this.state = {
        data: []
    };
}

P = _ReactWX2.default.toClass(P, _ReactWX2.default.Component, {
    componentDidMount: function () {
        let that = this;
        wx.request({
            url: 'http://yapi.demo.qunar.com/mock/18752/qunar/city',
            success: function (data) {
                let curData = that.cleanData(data.data);
                console.log(curData);
                that.setState({ data: curData });
            }
        });
    },
    cleanData: function (data) {
        let result = [];
        let obj = {};

        data.map(item => {
            if (/[A-Z]/.test(item)) {
                if (item !== 'A') {
                    result.push(obj);
                }
                obj = {};
                obj.title = item;
                obj.data = [];
            } else {
                obj.data.push(item);
            }
        });
        result.push(obj);
        return result;
    },
    render: function () {
        var h = _ReactWX2.default.createElement;

        return h('view', { 'class': 'city-select' }, this.state.data.map(function (el, i1060) {
            return h('view', null, el.title, '-------', h('view', null, el.data.map(function (elem, i1200) {
                return h('view', null, elem);
            }, this)));
        }, this));;
    },
    classUid: 'c1461'
}, {});
Page(_ReactWX2.default.toPage(P, 'pages/demo/syntax/test/index'));

exports.default = P;
