// anujs/aaa/dist/pages/demo/syntax/redux/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _ReactWX = require('../../../../ReactWX');

var _ReactWX2 = _interopRequireDefault(_ReactWX);

var _redux = require('../../../../npm/redux/lib/redux.js');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

let reducer = function (state = { list: [] }, action) {
    switch (action.type) {
        case 'ADD_ITEM':
            state.list.push(action.payload);
            return {
                list: [...state.list]
            };
        default:
            return state;
    }
};

let STORE = (0, _redux.createStore)(reducer);

// eslint-disable-next-line

function P() {
    this.state = {
        list: []
    };
}

P = _ReactWX2.default.toClass(P, _ReactWX2.default.Component, {
    componentDidMount: function () {
        var _self = this;
        STORE.subscribe(function () {
            _self.setState({
                list: STORE.getState().list
            });
        });
    },
    add: function () {

        STORE.dispatch({
            type: 'ADD_ITEM',
            payload: {
                id: +new Date(),
                name: '张三'
            }
        });
    },
    render: function () {
        var h = _ReactWX2.default.createElement;

        return h('view', { 'class': 'page' }, h('view', null, this.state.list.map(function (item, i1184) {
            return h('view', null, item.name, ' : ', item.id);
        }, this)), h('button', { onTap: this.add, 'data-tap-uid': 'e2770', 'data-class-uid': 'c1847', 'data-instance-uid': this.props.instanceUid }, 'click me'));;
    },
    classUid: 'c1847'
}, {});
Page(_ReactWX2.default.toPage(P, 'pages/demo/syntax/redux/index'));

exports.default = P;
