// anujs/aaa/src/pages/demo/syntax/request/index.js
import React from '@react';

class P extends React.Component {
    constructor() {
        super();
        this.state = {
            data: []
        };
    }
    componentDidMount() {
        let that = this;
        wx.request({
            url: 'http://yapi.demo.qunar.com/mock/18752/qunar/city',
            success: function(data) {
                let curData = that.cleanData(data.data);
                console.log(curData);
                that.setState({data: curData});
            }
        });
    }
    cleanData(data) {
        let result = [];
        let obj = {};

        data.map((item)=> {
            if (/[A-Z]/.test(item)) {
                if (item !== 'A') {
                    result.push(obj);
                }
                obj = {};
                obj.title = item;
                obj.data = [];
            } else {
                obj.data.push(item);
            }
        });
        result.push(obj);
        return result;
    }
    render() {
        return (
            <div class='city-select'>
                {this.state.data.map(function(el){
                    return <p>{el.title}-------
                        <ul>
                            {el.data.map(function(elem){
                                return <p>{elem}</p>;
                            })}
                        </ul>
                    </p>;
                },this)}
            </div>
        );
    }
}

export default P;
